src/application/jobs/notifier-campagne.job.handler.db.ts

```diff
@@ -35,34 +36,43 @@ export class NotifierCJEJobHandler extends JobHandler<Job> {
     @Inject(PlanificateurRepositoryToken)
     private planificateurRepository: Planificateur.Repository
   ) {
-    super(Planificateur.JobType.NOTIFIER_CJE, suiviJobService)
+    super(Planificateur.JobType.NOTIFIER_CAMPAGNE, suiviJobService)
   }
 
   async handle(
-    job?: Planificateur.Job<Planificateur.JobNotifierParGroupe>
+    job: Planificateur.Job<{
+      offset: number
+      idCampagne: string
+      nbNotifsEnvoyees: number
+    }>
   ): Promise<SuiviJob> {
     let succes = true
     const stats: Stats = {
-      nbPersonnesNotifiees: job?.contenu?.nbPersonnesNotifiees || 0,
+      nbNotifsEnvoyees: job.contenu.nbNotifsEnvoyees,
       estLaDerniereExecution: false
     }
     const maintenant = this.dateService.now()
 
     try {
-      const structuresConcernees = [
-        Core.Structure.MILO,
-        Core.Structure.POLE_EMPLOI
-      ]
+      const offset = job.contenu.offset
 
-      const offset = job?.contenu?.offset || 0
+      const idsJeunesQuiOntReponduALaCampagne = (
+        await ReponseCampagneSqlModel.findAll({
+          where: { idCampagne: job.contenu.idCampagne },
+          attributes: ['idJeune']
+        })
+      ).map(campagneSql => campagneSql.idJeune)
 
       const idsJeunesANotifier = await JeuneSqlModel.findAll({
         where: {
           structure: {
-            [Op.in]: structuresConcernees
+            [Op.in]: structuresCampagnes
           },
           pushNotificationToken: {
             [Op.ne]: null
+          },
+          id: {
+            [Op.notIn]: idsJeunesQuiOntReponduALaCampagne
```

Comment:
C'est hyper lourd nan un `notIn` avec des grosses listes ?

Reply:
id est indexé... mais you've got a point

Reply:
Piste : "NOT EXISTS"
https://sqlperformance.com/2012/12/t-sql-queries/left-anti-semi-join